python/dgl/distributed/rpc_client.py

```diff
@@ -187,16 +194,44 @@ def connect_to_server(ip_config, num_servers, max_queue_size=MAX_QUEUE_SIZE, net
     atexit.register(exit_client)
     set_initialized(True)
 
-def shutdown_servers():
+def shutdown_servers(ip_config, num_servers):
     """Issue commands to remote servers to shut them down.
 
+    This function is required to be called manually only when we
+    have booted servers which keep alive even clients exit. In
```

Comment:
is it right that this function is called in the process that tries to shut down all servers?

here are my questions:
* if the servers are launched with keepalive=False, how to shut them down?
* if the servers are launched with keepalive=True, does a trainer need to do anything before exits?

Reply:
if `keep_alive=True`,  if servers are launched manually, `dgl.distributed.shutdown_servers()` should be called just like what we do in `test_xxx.py`. If servers are launched with `tools/launch.py`, we could terminate the server processes directly or call `dgl.distributed.shutdown_servers()`.

if `keep_alive=False`, everything is exactly same as before. servers will exit once client_0 exit.